packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -1191,7 +1191,9 @@ class _NestedScrollController extends ScrollController {
 
   @override
   void attach(ScrollPosition position) {
-    assert(position is _NestedScrollPosition);
+    if (position is! _NestedScrollPosition) {
```

Comment:
does that mean the scroll position will not be attached to anything even if it attempt to attach? I think the right fix would be the nestedScrollPosition don't reset other scroll position in the sub tree even if it reaches the top

Reply:
No, the expected behavior is that it should not attach regardless of whether it reaches the top. This fix is correct because sharing the scroll position with other tabs, even when not at the top, is not natural behavior.

Reply:
Please check this related issue as well.
ref: https://github.com/flutter/flutter/issues/62833